internal/compiler/utils.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package compiler
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/util"
+)
+
+func VerifyAuthorizationPolicySchema(input map[string]*ast.Module, ref ast.Ref) error {
+	modules := map[string]*ast.Module{}
+
+	for name, module := range input {
+		for _, rule := range module.Rules {
+			if rule.Ref().Equal(ref) {
+				modules[name] = module
```

Comment:
This will pick up non-matching rules in the module and apply the schema on them as well. This needs to be updated as it's unintended.